core/trino-main/src/main/java/io/trino/execution/DynamicFilterConfig.java

```diff
@@ -130,244 +125,127 @@ public DynamicFilterConfig setDynamicRowFilterSelectivityThreshold(double dynami
     }
 
     @Min(0)
-    public int getSmallMaxDistinctValuesPerDriver()
-    {
-        return smallMaxDistinctValuesPerDriver;
-    }
-
-    @LegacyConfig("dynamic-filtering.small-broadcast.max-distinct-values-per-driver")
-    @Config("dynamic-filtering.small.max-distinct-values-per-driver")
-    public DynamicFilterConfig setSmallMaxDistinctValuesPerDriver(int smallMaxDistinctValuesPerDriver)
-    {
-        this.smallMaxDistinctValuesPerDriver = smallMaxDistinctValuesPerDriver;
-        return this;
-    }
-
-    @MaxDataSize("1MB")
-    public DataSize getSmallMaxSizePerDriver()
-    {
-        return smallMaxSizePerDriver;
-    }
-
-    @LegacyConfig("dynamic-filtering.small-broadcast.max-size-per-driver")
-    @Config("dynamic-filtering.small.max-size-per-driver")
-    public DynamicFilterConfig setSmallMaxSizePerDriver(DataSize smallMaxSizePerDriver)
-    {
-        this.smallMaxSizePerDriver = smallMaxSizePerDriver;
-        return this;
-    }
-
-    @Min(0)
-    public int getSmallRangeRowLimitPerDriver()
-    {
-        return smallRangeRowLimitPerDriver;
-    }
-
-    @LegacyConfig("dynamic-filtering.small-broadcast.range-row-limit-per-driver")
-    @Config("dynamic-filtering.small.range-row-limit-per-driver")
-    public DynamicFilterConfig setSmallRangeRowLimitPerDriver(int smallRangeRowLimitPerDriver)
-    {
-        this.smallRangeRowLimitPerDriver = smallRangeRowLimitPerDriver;
-        return this;
-    }
-
-    @MaxDataSize("10MB")
-    public DataSize getSmallMaxSizePerOperator()
-    {
-        return smallMaxSizePerOperator;
-    }
-
-    @LegacyConfig("dynamic-filtering.small-broadcast.max-size-per-operator")
-    @Config("dynamic-filtering.small.max-size-per-operator")
-    public DynamicFilterConfig setSmallMaxSizePerOperator(DataSize smallMaxSizePerOperator)
-    {
-        this.smallMaxSizePerOperator = smallMaxSizePerOperator;
-        return this;
-    }
-
-    @Min(0)
-    public int getSmallPartitionedMaxDistinctValuesPerDriver()
-    {
-        return smallPartitionedMaxDistinctValuesPerDriver;
-    }
-
-    @Config("dynamic-filtering.small-partitioned.max-distinct-values-per-driver")
```

Comment:
Shouldn't there be a deprecation period during which config still works but using it triggers a warning?

cc @kokosing